test/functional/test_framework/test_framework.py

```diff
@@ -495,6 +496,10 @@ def get_bin_from_version(version, bin_name, bin_default):
             extra_confs = [[]] * num_nodes
         if extra_args is None:
             extra_args = [[]] * num_nodes
+        # Whitelist peers to speed up tx relay / mempool sync. Don't use it if testing tx relay or timing.
+        if self.whitelist_peers:
+            for i in range(len(extra_args)):
+                extra_args[i] = extra_args[i] + ["-whitelist=noban,in,out@127.0.0.1"]
```

Comment:
I'm not sure this is sufficient functional test coverage. With the exception of the new permissions "in" and "out" which are invalid on master, all tests otherwise pass with this commit (1e09c265a9598df3cc39336e3bcccb993dacf3d8) applied.
I was expecting to see a test that makes an outbound connection to a node that misbehaves and doesn't get banned.

Or was the motivation more about the tx "trickle"? In that case I can see it being hard to test without waiting for time to pass?